airbyte-db/db-lib/src/main/java/io/airbyte/db/IncrementalUtils.java

```diff
@@ -15,7 +15,10 @@ public class IncrementalUtils {
 
   @SuppressWarnings("PMD.AvoidLiteralsInIfCondition")
   public static String getCursorField(final ConfiguredAirbyteStream stream) {
-    if (stream.getCursorField().size() == 0) {
+    if (stream.getDataToSyncConfig().replication_method == ReplicationMethod.CDC) {
+      // FIXME: This isn't the best place for this test to happen. Move to a proper validation method.
```

Comment:
Move to: `logPreSyncDebugData()`

Reply:
Update: after reviewing `logPreSyncDebugData()` and with some encouragement from @rodireich to suggest additional changes as needed, I wanted to propose here that we add a higher level method: `validateSyncSettings(database, catalog, state)` which calls `logPreSyncDebugData()` and also calls a new `checkStateAndCatalogCompatibility(database, catalog, state)`.

@rodireich - Any initial thoughts on this approach?

Reply:
I think that's the place I would do this check also.
One thing is that the name may create confusion with the check() in protocol which I cannot remember if runs every time you kick off a sync or just when making config or catalog changes.

Reply:
> One thing is that the name may create confusion with the check() in protocol

I thought about this as well. While I do think it would be helpful for this to be performed in the `check` operation, logistically, I don't think that can happen today because `check` only takes `config` does not take in the catalog and state.